pyintegration/deephaven2/time.py

```diff
@@ -0,0 +1,830 @@
+
+
+class TimeZone(Enum):
+    """ A Enum for known time zones. """
+    NY = _JTimeZone.TZ_NY
+    """ America/New_York """
+    ET = _JTimeZone.TZ_ET
+    """ America/New_York """
+    MN = _JTimeZone.TZ_MN
+    """ America/Chicago """
+    CT = _JTimeZone.TZ_CT
+    """ America/Chicago """
+    MT = _JTimeZone.TZ_MT
+    """ America/Denver """
+    PT = _JTimeZone.TZ_PT
+    """ America/Los_Angeles """
+    HI = _JTimeZone.TZ_HI
+    """ Pacific/Honolulu """
+    BT = _JTimeZone.TZ_BT
+    """ America/Sao_Paulo """
+    KR = _JTimeZone.TZ_KR
+    """ Asia/Seoul """
+    HK = _JTimeZone.TZ_HK
+    """ Asia/Hong_Kong """
+    JP = _JTimeZone.TZ_JP
+    """ Asia/Tokyo """
+    AT = _JTimeZone.TZ_AT
+    """ Canada/Atlantic """
+    NF = _JTimeZone.TZ_NF
+    """ Canada/Newfoundland """
+    AL = _JTimeZone.TZ_AL
+    """ America/Anchorage """
+    IN = _JTimeZone.TZ_IN
+    """ Asia/Kolkata """
+    CE = _JTimeZone.TZ_CE
+    """ Europe/Berlin """
+    SG = _JTimeZone.TZ_SG
+    """ Asia/Singapore """
+    LON = _JTimeZone.TZ_LON
+    """ Europe/London """
+    MOS = _JTimeZone.TZ_MOS
+    """ Europe/Moscow """
+    SHG = _JTimeZone.TZ_SHG
+    """ Asia/Shanghai """
+    CH = _JTimeZone.TZ_CH
+    """ Europe/Zurich """
+    NL = _JTimeZone.TZ_NL
+    """ Europe/Amsterdam """
+    TW = _JTimeZone.TZ_TW
+    """ Asia/Taipei """
+    SYD = _JTimeZone.TZ_SYD
+    """ Australia/Sydney """
+    UTC = _JTimeZone.TZ_UTC
+    """ UTC """
+
+
+def to_datetime(s: str, quiet: bool = False) -> DateTime:
+    """ Converts a datetime string to a DateTime object.
+
+    Args:
+        s (str): in the form of "yyyy-MM-ddThh:mm:ss[.SSSSSSSSS] TZ"
+        quiet (bool): when True, if the datetime string can't be parsed, this function returns None, otherwise
+            it raises an exception. The default is False
+
+    Returns:
+        a DateTime
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertDateTimeQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertDateTime(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def to_period(s: str, quiet: bool = False) -> Period:
+    """ Converts a period string into a Period object.
+
+    Args:
+        s (str): a string in the form of nYnMnWnDTnHnMnS, with n being numeric values, e.g. 1W for one week, T1M for
+            one minute, 1WT1H for one week plus one hour
+        quiet (bool): when True, if the period string can't be parsed, this function returns None, otherwise
+            it raises an exception. The default is False
+
+    Returns:
+        a Period
+
+    Raises:
+        DHError
+    """
+    if quiet:
+        return _JDateTimeUtils.convertPeriodQuiet(s)
+
+    try:
+        return _JDateTimeUtils.convertPeriod(s)
+    except Exception as e:
+        raise DHError(e) from e
+
+
+def to_nanos(s, quiet: bool = False) -> int:
+    """ Converts a time string to nanoseconds from the Epoch.
```

Comment:
I think this description is wrong.  I think we are taking a string like "01:02:03" and converting it to HOUR+2*MINUTE+3*SECOND.  This is not time since the Epoch.  Am I wrong?

Reply:
you are right. The description is wrong, just straight conversion not from Epoch.